sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -563,10 +668,13 @@ class SpillableColumnarBatchHandle private (
   }
 
   override def close(): Unit = {
-    releaseDeviceResource()
     synchronized {
+      if (toSpill.isEmpty) {
```

Comment:
Why is this different from the other resources? Like host resources?

Reply:
We were observing double frees which traced back to a race here specifically for device columnar batches, but you're right that we should check for host resources too.